framework/wazuh/core/indexer/__init__.py

```diff
@@ -0,0 +1,120 @@
+import os
+import random
+from asyncio import sleep
+from logging import getLogger
+
+from opensearchpy import AsyncOpenSearch
+from wazuh.core.exception import WazuhIndexerError
+
+logger = getLogger('wazuh')
+
+HOST_KEY = 'host'
+PORT_KEY = 'port'
+
+
+class Indexer:
+    """Interface to connect with Wazuh Indexer."""
+
+    def __init__(
+        self,
+        host: str,
+        user: str,
+        password: str,
+        port: int = 9200,
+        use_ssl: bool = True,
+        verify_certs: bool = True,
+        ca_certs: str = '',
+    ) -> None:
+        self.host = host
+        self.user = user
+        self.password = password
+        self.port = port
+        self.use_ssl = use_ssl
+        self.verify_certs = verify_certs
+        self.ca_certs = ca_certs
+
+        self._client = self._get_opensearch_client()
+
+        # Register index clients here
+
+    def _get_opensearch_client(self) -> AsyncOpenSearch:
+        """Get the a new instance of the opensearch client.
+
+        Returns
+        -------
+        AsyncOpenSearch
+            The created instance.
+        """
+        return AsyncOpenSearch(
+            hosts=[{HOST_KEY: self.host, PORT_KEY: self.port}],
+            http_compress=True,
+            http_auth=(self.user, self.password),
+            use_ssl=self.use_ssl,
+            verify_certs=self.verify_certs,
+            ca_certs=self.ca_certs,
+        )
+
+    async def initialize(self):
```

Comment:
I would prefer `connect`, since we a re not modifying any internal variables

```suggestion
    async def connect(self) -> None:
```